RedisMetadataReport password not fetch correct value  from url

- [x] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://github.com/apache/dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 2.7.3
* Operating System version: macOS
* Java version: 1.8

### Steps to reproduce this issue

use redis as a metadata storage, and redis server should password to auth


```
at org.apache.dubbo.metadata.store.redis.RedisMetadataReport.storeMetadataInCluster(RedisMetadataReport.java:92)
	at org.apache.dubbo.metadata.store.redis.RedisMetadataReport.storeMetadata(RedisMetadataReport.java:83)
	at org.apache.dubbo.metadata.store.redis.RedisMetadataReport.doStoreProviderMetadata(RedisMetadataReport.java:71)
	at org.apache.dubbo.metadata.support.AbstractMetadataReport.storeProviderMetadataTask(AbstractMetadataReport.java:245)
	at org.apache.dubbo.metadata.support.AbstractMetadataReport.lambda$storeProviderMetadata$0(AbstractMetadataReport.java:232)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: redis.clients.jedis.exceptions.JedisDataException: NOAUTH Authentication required.
	at redis.clients.jedis.Protocol.processError(Protocol.java:127)
	at redis.clients.jedis.Protocol.process(Protocol.java:161)
	at redis.clients.jedis.Protocol.read(Protocol.java:215)
	at redis.clients.jedis.Connection.readProtocolWithCheckingBroken(Connection.java:340)
	at redis.clients.jedis.Connection.getRawObjectMultiBulkReply(Connection.java:285)
	at redis.clients.jedis.Connection.getObjectMultiBulkReply(Connection.java:291)
	at redis.clients.jedis.Jedis.clusterSlots(Jedis.java:3376)
	at redis.clients.jedis.JedisClusterInfoCache.discoverClusterNodesAndSlots(JedisClusterInfoCache.java:54)
	at redis.clients.jedis.JedisClusterConnectionHandler.initializeSlotsCache(JedisClusterConnectionHandler.java:39)
	at redis.clients.jedis.JedisClusterConnectionHandler.<init>(JedisClusterConnectionHandler.java:17)
	at redis.clients.jedis.JedisSlotBasedConnectionHandler.<init>(JedisSlotBasedConnectionHandler.java:24)
	at redis.clients.jedis.BinaryJedisCluster.<init>(BinaryJedisCluster.java:54)
	at redis.clients.jedis.JedisCluster.<init>(JedisCluster.java:93)
	at org.apache.dubbo.metadata.store.redis.RedisMetadataReport.storeMetadataInCluster(RedisMetadataReport.java:88)
	... 7 more
```

--- a/dubbo-metadata-report/dubbo-metadata-report-redis/src/main/java/org/apache/dubbo/metadata/store/redis/RedisMetadataReport.java
+++ b/dubbo-metadata-report/dubbo-metadata-report-redis/src/main/java/org/apache/dubbo/metadata/store/redis/RedisMetadataReport.java
@@ -56,6 +56,7 @@ public class RedisMetadataReport extends AbstractMetadataReport {
     public RedisMetadataReport(URL url) {
         super(url);
         timeout = url.getParameter(TIMEOUT_KEY, DEFAULT_TIMEOUT);
+        password = url.getPassword();
         if (url.getParameter(CLUSTER_KEY, false)) {
             jedisClusterNodes = new HashSet<HostAndPort>();
             List<URL> urls = url.getBackupUrls();
@@ -64,7 +65,7 @@ public class RedisMetadataReport extends AbstractMetadataReport {
             }
         } else {
             int database = url.getParameter(REDIS_DATABASE_KEY, 0);
-            pool = new JedisPool(new JedisPoolConfig(), url.getHost(), url.getPort(), timeout, url.getPassword(), database);
+            pool = new JedisPool(new JedisPoolConfig(), url.getHost(), url.getPort(), timeout, password, database);
         }
     }
 

